File: sdk/php/src/Resources/VCloudStack.php
<?php

namespace HiTechCloud\SDK\Resources;

class VCloudStack extends BaseResource
{
    /**
     * Rescue VM
     */
    public function rescue(int $serviceId, int $vmId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/vms/' . $vmId . '/rescue');
    }

    /**
     * Unrescue VM
     */
    public function unrescue(int $serviceId, int $vmId): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/vms/' . $vmId . '/unrescue');
    }

    /**
     * Get console URL
     */
    public function getConsole(int $serviceId, int $vmId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/vms/' . $vmId . '/console');
    }

    /**
     * Get VM usage
     */
    public function getUsage(int $serviceId, int $vmId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/vms/' . $vmId . '/usage');
    }

}

File: sdk/csharp/Resources/Vcloudstack.cs
namespace HiTechCloud.SDK.Resources;

public class Vcloudstack : BaseResource
{
    public Vcloudstack(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> RescueAsync(int id, int vmid)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/vms/{vmid}/rescue", id, vmid), null);
    }

    public async Task<Dictionary<string, object?>> UnrescueAsync(int id, int vmid)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/vms/{vmid}/unrescue", id, vmid), null);
    }

    public async Task<Dictionary<string, object?>> ConsoleAsync(int id, int vmid)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/vms/{vmid}/console", id, vmid));
    }

    public async Task<Dictionary<string, object?>> UsageAsync(int id, int vmid)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/vms/{vmid}/usage", id, vmid));
    }
}
